internal/config/variables/variables.go

```diff
@@ -666,8 +680,71 @@ func EvaluateVariables(
 			})
 		}
 	}
+	// Error here if we have them from parsing
+	if diags.HasErrors() {
+		return nil, nil, diags
+	}
+
+	jobVals, diags := getJobValues(vs, iv, salt)
+	if diags.HasErrors() {
+		return nil, nil, diags
+	}
+	return iv, jobVals, diags
+}
+
+// getJobValues combines the Variable and Value into a VariableRef,
+// encoding any 'sensitive' values as SHA256 values with the given salt.
```

Comment:
```suggestion
// hashing any 'sensitive' values as SHA256 values with the given salt.
```
Sorry super nit, "encoding" & "hashing" aren't the same. Encoding is reversible and not a security mechanism, hashing is a security tool and irreversible.